### PR TITLE
Request timeout settings for the same domain name are reused

### PR DESCRIPTION
**Pull request by byte0o**
fix client http SetReadDeadline/SetWriteDeadline Deadline is reused

### CODE REVIEW

**Review by erikdubbelboer — CHANGES_REQUESTED** [review]
Great find!

It seems like you have to run gofmt to format the code properly. You're using spaces now but Go code is always indented with tabs.

The comments also aren't needed.

And it seems like some tests are failing now because they don't expect these functions to be called. Can you look into that?

File: client.go [hunk 1]
```diff
@@ -1380,13 +1380,11 @@ func (c *HostClient) doNonNilReqResp(req *Request, resp *Response) (bool, error)
...
 	}
-	if !writeDeadline.IsZero() {
-		// Set Deadline every time, since golang has fixed the performance issue
-		// See https://github.com/golang/go/issues/15133#issuecomment-271571395 for details
-		if err = conn.SetWriteDeadline(writeDeadline); err != nil {
-			c.closeConn(cc)
-			return true, err
-		}
+	
+	// fix Deadline is reused
```
[the remark this answers is not in the document]
Comment (byte0o) [reply client.go]: done

File: client.go [hunk 2]
```diff
@@ -1425,13 +1423,11 @@ func (c *HostClient) doNonNilReqResp(req *Request, resp *Response) (bool, error)
...
 	}
-	if !readDeadline.IsZero() {
-		// Set Deadline every time, since golang has fixed the performance issue
-		// See https://github.com/golang/go/issues/15133#issuecomment-271571395 for details
-		if err = conn.SetReadDeadline(readDeadline); err != nil {
-			c.closeConn(cc)
-			return true, err
-		}
+	
+	// fix Deadline is reused
```
[the remark this answers is not in the document]
Comment (byte0o) [reply client.go]: done

[the remark this answers is not in the document]
Comment (byte0o) [reply client.go]: The test failed because the tested Conn did not implement the correct SetWriteDeadline/SetReadDeadline func,Already fixed

**Comment by erikdubbelboer** [comment]
Thanks!

**Comment by ReneWerner87** [comment]
i have tested fiber with the master version and found this error

image
image

image

do i have to adjust anything else for us or is it a real error that has not been noticed ? 
@byte0o @erikdubbelboer 

https://github.com/gofiber/fiber/blob/9effdf829a79df49c24318bf753f8332d6e75375/client_test.go#L608-L642

**Comment by erikdubbelboer** [comment]
@ReneWerner87 as you can see from the stacktrace the errors is in `(*readErrorConn).SetWriteDeadline` which doesn't exist in https://github.com/gofiber/fiber/blob/master/client_test.go#L584 so I'm not sure which code you are running? But the issue isn't in fasthttp.

**Comment by ReneWerner87** [comment]
Ok I see, so we need to add these 2 methods to our test connection as well

Thank you for the clarification
(have a nice weekend)

**Pull request merged**